### PR TITLE
Standardize "requires a subcommand" error message

### LINKED ISSUE #298: "requires a subcommand" error without examples
I noticed an inconsistency in the help message when using commands that require subcommands. When I call the [`list` command in my `friends` CLI app](https://github.com/JacobEvelyn/friends/blob/master/lib/friends/commands/list.rb#L4), it gives examples:

```
friends list
Error: Command 'list' requires a subcommand friends,activities,notes,locations,tags,favorite
```

but when I use [one of those subcommands that also requires further subcommands](https://github.com/JacobEvelyn/friends/blob/master/lib/friends/commands/list.rb#L94), I see no examples:

```
friends list favorite
Error: Command 'favorite' requires a subcommand
```

Is there a way to get that second example to also show the subcommand options?

### PR DESCRIPTION
**Pull request by JacobEvelyn**
Closes #298

### CODE REVIEW

File: lib/gli/command_support.rb [hunk 1]
```diff
@@ -166,14 +166,10 @@ def get_action(arguments)
...
     def generate_error_action(arguments)
       lambda { |global_options,options,arguments|
-        if am_subcommand?
-          if arguments.size > 0
-            raise UnknownCommand,"Unknown command '#{arguments[0]}'"
-          else
-            raise BadCommandLine,"Command '#{name}' requires a subcommand"
-          end
-        elsif have_subcommands?
+        if have_subcommands?
```
Comment (JacobEvelyn) [inline lib/gli/command_support.rb:169]: I'm pretty sure that reordering these conditionals is safe but please double-check that!

Comment (davetron5000) [reply lib/gli/command_support.rb:169]: It seems to be breaking this test:

https://github.com/davetron5000/gli/blob/main/test/tc_subcommands.rb#L149

(sorry about the weird test framework—Given/When/Then don't do anything so the test is vanilla Test::Unit really)

That test is wanting to see that if the command has a subcommand, but extra args were given, the message is "Unknown command".  I guess that error check has precedence, so I think this `if` statement does need to be kinda (annoyingly) complex.

Comment (JacobEvelyn) [reply lib/gli/command_support.rb]: Ah whoops! I think I can just reorder the `if` and `elsif` parts of the conditional—let me know if that seems right now.

(I'm trying out GitHub's "make a code change and PR right from the browser" feature but a downside is I can't run tests. Let me know if this process is too annoying for you.)

**Comment by davetron5000** [comment]
This version passes the tests. I don't feel strongly about having a failing test that this PR fixes since it's mostly about wording and the code gets executed by the tests.  Thanks!

**Comment by davetron5000** [comment]
released as 2.19.2

**Comment by JacobEvelyn** [comment]
Thank you!

**Pull request merged**